### DISCUSSION TITLE
need a cheap way to get  the current undo block info

### DISCUSSION THREAD

**Discussion by haolian9** [neovim/neovim#24594] [General]
### Problem

hi folks!
currently the only way is using fn.undotree(), while it's relatively heavy as it contains the whole undo tree, especially i trend to have a big &undolevels, say 1000.

assume i have such mapping `nnoremap gq my_format` to trigger a format operation, and input `gq u <c-r> gq` occasionally.
i want to avoid my_format() being ran at the second `gq`, as `u <c-r>` implies there is no changes.
and `b:changedtick` wont help here, since `u` and `<c-r>` always increase it.

so i need a way to get the undo block info after `gq`, and compare it with the one when i raise gq again.
also from what i know `undo!` can clear part of the undo history, so i should use seq and time togger.

the problem originally came from [here](https://github.com/nvimdev/guard.nvim/issues/27)

### Expected behavior

a cheap way to get the current undo block info

**Comment by haolian9** [comment]
obvisiouly this interested nobody, for myself it's a rare usecase, and being not familiar to the code, i dont think it's worth it in the end. so i'm closing it.

**Reply by bew** [reply]
I think it definitely has value, could you re-open ?

Having the ability to get the last n undo block info, or even have an iterator over undo block would be nice and cheap
